### PR TITLE
[ci] Enable the Style/ConditionalAssignment cop in Rubocop

### PR DESCRIPTION
**Pull request by DavidKang**
Check for if and case statements where each branch is used for assignment
to the same variable when using the return of the condition can be used
instead.

### CODE REVIEW

**Comment by coolo** [comment]
IMHO this makes understanding code much harder.

File: src/api/app/controllers/message_controller.rb [hunk 1]
```diff
@@ -17,12 +17,12 @@ def check_project_and_package
   end
 
   def list
-    if @package
-      @messages = @package.messages
+    @messages = if @package
```
Comment (Ana06) [inline src/api/app/controllers/message_controller.rb]: strange indentation, wouldn't it be better to have `if`, `elseif` and `else` aligned?

File: src/api/.rubocop.yml [hunk 2]
```diff
@@ -105,6 +105,10 @@ Style/TernaryParentheses:
 Style/MethodCallParentheses:
   Enabled: true
 
+# Check for if and case statements where each branch is used for assignment
```
Comment (Ana06) [inline src/api/.rubocop.yml]: And we should order this file alphabetically again!! :sob: (although unrelated to the PR)

File: src/api/app/controllers/person_controller.rb [hunk 3]
```diff
@@ -11,10 +11,10 @@ class PersonController < ApplicationController
   skip_before_action :require_login, only: [:command, :register]
 
   def show
-    if params[:prefix]
-        @list = User.where("login LIKE ?", params[:prefix] + '%')
+    @list = if params[:prefix]
+        User.where("login LIKE ?", params[:prefix] + '%')
     else
```
Comment (Ana06) [inline src/api/app/controllers/person_controller.rb]: strange indentation, wouldn't it be better to have `if` and `else` aligned?

File: src/api/app/controllers/search_controller.rb [hunk 4]
```diff
@@ -185,10 +185,10 @@ def search(what, render_all)
     output = "<collection matches=\"#{matches}\">\n"
 
     xml = Hash.new # filled by filter
-    if render_all
-      key_template = "xml_#{what}_%d"
+    key_template = if render_all
```
Comment (Ana06) [inline src/api/app/controllers/search_controller.rb]: strange indentation, wouldn't it be better to have `if` and `else` aligned?

File: src/api/app/controllers/search_controller.rb [hunk 5]
```diff
@@ -269,16 +269,16 @@ def find_attribute(namespace, name)
 
     # gather the relation for attributes depending on project/package combination
     if params[:package]
-      if params[:project]
-        attribs = Package.get_by_project_and_name(params[:project], params[:package]).attribs
+      attribs = if params[:project]
```
Comment (Ana06) [inline src/api/app/controllers/search_controller.rb]: strange indentation, wouldn't it be better to have `if`, `elseif` and `else` aligned?

File: src/api/app/controllers/search_controller.rb [hunk 6]
```diff
@@ -269,16 +269,16 @@ def find_attribute(namespace, name)
...
+        Package.get_by_project_and_name(params[:project], params[:package]).attribs
       else
-        attribs = attrib.attribs.where(package_id: Package.where(name: params[:package]))
+        attrib.attribs.where(package_id: Package.where(name: params[:package]))
       end
     else
-      if params[:project]
-        attribs = attrib.attribs.where(package_id: Project.get_by_name(params[:project]).packages)
+      attribs = if params[:project]
```
Comment (Ana06) [inline src/api/app/controllers/search_controller.rb]: strange indentation, wouldn't it be better to have `if` and `else` aligned?

File: src/api/app/controllers/statistics_controller.rb [hunk 7]
```diff
@@ -28,10 +28,10 @@ def highest_rated
     ratings = Rating.select('db_object_id, db_object_type, count(score) as count,' +
                                 'sum(score)/count(score) as score_calculated').group('db_object_id, db_object_type').order('score_calculated DESC')
     ratings = ratings.to_a.delete_if { |r| r.count.to_i < min_votes_for_rating }
-    if @limit
-      @ratings = ratings[0..@limit-1]
+    @ratings = if @limit
```
Comment (Ana06) [inline src/api/app/controllers/statistics_controller.rb]: strange indentation, wouldn't it be better to have `if` and `else` aligned?

File: src/api/app/controllers/statistics_controller.rb [hunk 8]
```diff
@@ -131,10 +131,10 @@ def latest_added
     list.concat packages
     list.sort! { |a, b| b.created_at <=> a.created_at }
 
-    if @limit
-      @list = list[0..@limit-1]
+    @list = if @limit
```
Comment (Ana06) [inline src/api/app/controllers/statistics_controller.rb]: strange indentation, wouldn't it be better to have `if` and `else` aligned?

**Comment by Ana06** [comment]
@coolo 

> IMHO this makes understanding code much harder.

why?

**Comment by Ana06** [comment]
There is another option in the cop which enforce you to do it just the other way around, assigning inside condition. So if we can not agree on enable this rule like it is now, I would like to enable it with the other condition, to be consistent with the style we want to use at least.

**Comment by coolo** [comment]
Ana, I think that if you make a long statement where you do not express well enough your intentions with and where you have to look up at the beginning of the statement how it started and it is mixed with a lot of other content that are completely unrelated to the original intention (like talking about the weather or daily politics - e.g. budget planing reports in the middle of a pull request comment) you make it harder to understand than simple statements. You know, I like plain statements. I told you before. Not always. But most of the time I do.

File: src/api/app/controllers/message_controller.rb (hunk 1 above)
Comment (DavidKang) [inline src/api/app/controllers/message_controller.rb]: @Ana06, rubocop told me to align elsif with @messages, if you prefer I can put it like:
``` ruby
@message =
  if @package
    ...
  elsif @project
    ...
  end
```

Comment (DavidKang) [inline src/api/app/controllers/message_controller.rb]: @bgeuken, Is possible with [MultilineAssignment](http://rubocop.readthedocs.io/en/latest/cops_style/#stylemultilineassignmentlayout)

**Comment by DavidKang** [comment]
[Enable the Style/ConditionalAssignment cop in Rubocop](https://trello.com/c/JxHpnNwb/83-enable-the-style-conditionalassignment-cop-in-rubocop)

**Comment by ChrisBr** [comment]
@Ana06 I agree with @coolo, way harder to read!

**Comment by DavidKang** [comment]
I'm with @Ana06, but is true that actually is more hard to read, although with the proposed indentation (https://github.com/openSUSE/open-build-service/pull/2452#discussion_r91700486) it will be more readable

**Comment by hennevogel** [comment]
I also don't like it. I like the other option to enforce the opposite though  :-)

**Comment by bgeuken** [comment]
@DavidKang It doesn't even have to be like 
```ruby
@messages =
  if @package
    ...
  elsif @project
    ...
  end
```

Could also be 
```ruby
@messages = if @package
              ...
            elsif @project
              ... 
            end
```

And for sake of completeness
```ruby
if @package
  @messages =  ...
elsif @project
  @messages = ...
end
```

**Comment by DavidKang** [comment]
To admit this kind of alignment                                                     
                                                                                 
 ``` ruby                                                                        
 @messages = if @package                                                         
               ...                                                               
             elsif @project                                                      
               ...                                                               
             end                                                                 
 ```                                                                             
 we have to change [EndAlignment cop](http://rubocop.readthedocs.io/en/latest/cops_lint/#lintendalignment) to `keyword`, because actually in OBS it's set up with `variable`.
                                                                            
 I think that we can add this change in this PR.

**Comment by Ana06** [comment]
@DavidKang 


> we have to change EndAlignment cop to keyword

I like the idea. :+1:

**Comment by bgeuken** [comment]
>> DavidKang commented a day ago 
>>  we have to change EndAlignment cop to keyword
> Ana06 commented a day ago
>
> I like the idea. 👍

Me too. @openSUSE/open-build-service what do you guys say?

**Comment by ChrisBr** [comment]
I still prefer 
```
if @package
  @messages = ...
elsif @project
  @messages = ...
end
```

**Comment by DavidKang** [comment]
Then, @openSUSE/open-build-service, what should we do with the proposed changes? Apply them or not?

**Comment by hennevogel** [comment]
I guess we can close this and then reference this thread in a comment above the rule...

**Pull request closed without merging**